# Etape 1 : Exploration des données + DataViz’

Attendue de cette étape :

- Votre première tâche consistera à définir le contexte et le périmètre du projet : j’attends que vous preniez vraiment le temps de bien comprendre le projet et de vous renseigner au mieux sur les notions que celui-ci va introduire.

- Il faudra ensuite prendre en main et découvrir votre jeu de données et faire une analyse presque exhaustive de celui-ci afin de mettre en lumière la structure, les difficultés et eventuels biais du dataset.

J’attendrai également au moins 5 représentations graphiques construites à partir de votre jeu de données, visuelles et surtout pertinentes. Pour chacune d’elle j’attendrai :
- Un commentaire précis, qui analyse la figure et apporte un avis “métier”.
- Une validation du constat par des manipulations de données, ou un test statistique.

## Contexte et périmètre du projet

Le projet "Compagnon_Immo" a pour but de développer une solution innovante permettant aux acheteurs et investisseurs immobiliers d’explorer, comparer et estimer les prix des logements sur différents territoires.

Dans un premier temps nous nous focalisons sur l'analyse des données immobilières du Haut-Rhin (Département 68). Ces données comportent des informations sur des biens en locations et en ventes dans le département entre 2019 et 2023.

Nous avons à notre disposition une cinquentaine de variables pour prédire notre cible, le prix du bien.
Ces variables décrives les propriétés des biens comme leur surface, le nombre de pièces, d'étages etc ... mais aussi bien leur années de construction et leur emplacement géographique. Tout ces éléments impactent le prix du bien, tachons maintenant d'identifier les variables pertinantes pour notre analyse.

### Description des jeux de données

Voici un descriptif des variables disponibles dans nos jeux de données.

Jeu de données des locations dans le Haut-Rhin :

In [6]:
import pandas as pd

df_locations = pd.read_csv('data/ech_annonces_locations_68.csv', sep=';', index_col='idannonce')
df_ventes = pd.read_csv('data/ech_annonces_ventes_68.csv', sep=';', index_col='idannonce')

In [4]:
df_locations.columns.to_list()

['type_annonceur',
 'typedebien',
 'typedetransaction',
 'etage',
 'surface',
 'surface_terrain',
 'nb_pieces',
 'prix_bien',
 'prix_maison',
 'prix_terrain',
 'mensualiteFinance',
 'balcon',
 'eau',
 'bain',
 'dpeL',
 'dpeC',
 'mapCoordonneesLatitude',
 'mapCoordonneesLongitude',
 'annonce_exclusive',
 'nb_etages',
 'parking',
 'places_parking',
 'cave',
 'exposition',
 'ges_class',
 'annee_construction',
 'nb_toilettes',
 'nb_terraces',
 'videophone',
 'porte_digicode',
 'surface_balcon',
 'ascenseur',
 'nb_logements_copro',
 'charges_copro',
 'chauffage_energie',
 'chauffage_systeme',
 'chauffage_mode',
 'categorie_annonceur',
 'logement_neuf',
 'duree_int',
 'typedebien_lite',
 'date',
 'INSEE_COM',
 'IRIS',
 'CODE_IRIS',
 'TYP_IRIS_x',
 'TYP_IRIS_y',
 'GRD_QUART',
 'UU2010',
 'REG',
 'DEP']

In [11]:
df_ventes.columns.to_list()

['type_annonceur',
 'typedebien',
 'typedetransaction',
 'etage',
 'surface',
 'surface_terrain',
 'nb_pieces',
 'prix_bien',
 'prix_maison',
 'prix_terrain',
 'mensualiteFinance',
 'balcon',
 'eau',
 'bain',
 'dpeL',
 'dpeC',
 'mapCoordonneesLatitude',
 'mapCoordonneesLongitude',
 'annonce_exclusive',
 'nb_etages',
 'parking',
 'places_parking',
 'cave',
 'exposition',
 'ges_class',
 'annee_construction',
 'nb_toilettes',
 'nb_terraces',
 'videophone',
 'porte_digicode',
 'surface_balcon',
 'ascenseur',
 'nb_logements_copro',
 'charges_copro',
 'chauffage_energie',
 'chauffage_systeme',
 'chauffage_mode',
 'categorie_annonceur',
 'logement_neuf',
 'duree_int',
 'typedebien_lite',
 'date',
 'INSEE_COM',
 'IRIS',
 'CODE_IRIS',
 'TYP_IRIS_x',
 'TYP_IRIS_y',
 'GRD_QUART',
 'UU2010',
 'REG',
 'DEP',
 'loyer_m2_median_n6',
 'nb_log_n6',
 'taux_rendement_n6',
 'loyer_m2_median_n7',
 'nb_log_n7',
 'taux_rendement_n7',
 'prix_m2_vente']

In [59]:
df_locations[df_locations.columns[31:]].describe()

,nb_logements_copro,charges_copro,duree_int,INSEE_COM,IRIS,CODE_IRIS,GRD_QUART,UU2010,REG,DEP
count,9.000000,0.0,11448.000000,11702.000000,11702.000000,1.170200e+04,1.170200e+04,11702.000000,11702.000000,11702.000000
mean,18.666667,NaN,-207.466632,68191.909844,244.231499,6.819193e+08,6.819192e+06,68324.462400,44.018886,67.999316
std,42.272923,NaN,422.747918,814.069476,385.032725,8.140732e+06,8.140730e+04,3166.426849,1.763585,0.807832
min,0.000000,NaN,-1125.000000,6088.000000,0.000000,6.088010e+07,6.088010e+05,758.000000,11.000000,6.000000
25%,3.000000,NaN,-757.000000,68093.000000,0.000000,6.809300e+08,6.809300e+06,68402.000000,44.000000,68.000000
50%,5.000000,NaN,15.000000,68224.000000,103.000000,6.822402e+08,6.822401e+06,68501.000000,44.000000,68.000000
75%,8.000000,NaN,41.000000,68271.000000,202.000000,6.827101e+08,6.827101e+06,68701.000000,44.000000,68.000000
max,131.000000,NaN,504.000000,86214.000000,6220.000000,8.621401e+08,8.621401e+06,86601.000000,93.000000,86.000000


In [60]:
df_ventes[df_ventes.columns[31:]].describe()

,nb_logements_copro,charges_copro,duree_int,INSEE_COM,IRIS,CODE_IRIS,GRD_QUART,UU2010,REG,DEP,loyer_m2_median_n6,nb_log_n6,taux_rendement_n6,loyer_m2_median_n7,nb_log_n7,taux_rendement_n7,prix_m2_vente
count,7565.000000,6.677000e+03,26196.000000,27737.000000,27737.000000,2.773700e+04,2.773700e+04,27737.000000,27737.0,27737.0,17485.000000,17485.000000,17485.000000,19650.000000,19650.000000,19650.000000,27737.000000
mean,53.632386,1.805802e+03,-118.181898,68193.431734,172.923532,6.819345e+08,6.819344e+06,68370.913834,44.0,68.0,10.779532,17.011210,0.056380,10.639302,151.035929,0.056315,2624.974439
std,86.445999,3.155174e+04,390.366391,106.254100,336.887356,1.062528e+06,1.062542e+04,269.815128,0.0,0.0,2.066664,29.391785,0.083427,1.946427,252.454136,0.081040,1195.389487
min,0.000000,0.000000e+00,-1125.000000,68001.000000,0.000000,6.800100e+08,6.800100e+06,68000.000000,44.0,68.0,3.080000,1.000000,0.003000,4.780000,1.000000,0.003000,15.590000
25%,10.000000,6.600000e+02,6.000000,68084.000000,0.000000,6.808400e+08,6.808400e+06,68112.000000,44.0,68.0,9.440000,2.000000,0.038000,9.320000,3.000000,0.038000,1870.000000
50%,24.000000,1.200000e+03,22.000000,68222.000000,101.000000,6.822200e+08,6.822200e+06,68403.000000,44.0,68.0,10.530000,6.000000,0.048000,10.380000,18.000000,0.048000,2580.650000
75%,66.000000,2.000000e+03,77.000000,68279.000000,104.000000,6.827900e+08,6.827900e+06,68701.000000,44.0,68.0,11.950000,19.000000,0.066000,11.810000,190.000000,0.066000,3268.480000
max,3442.000000,2.577352e+06,793.000000,68386.000000,1802.000000,6.838600e+08,6.838600e+06,68701.000000,44.0,68.0,23.730000,212.000000,10.496000,23.730000,960.000000,10.773000,79500.000000


In [65]:
df_ventes.chauffage_energie.value_counts().index.sort_values()

Index(['Bois', 'Bois, Fioul', 'Bois, Électrique', 'Fioul', 'Fioul, Bois',
       'Gaz', 'Gaz, Bois', 'Gaz, Électrique', 'Électrique',
       'Électrique, Bois'],
      dtype='object', name='chauffage_energie')

In [66]:
df_locations.chauffage_energie.value_counts().index.sort_values()

Index(['Bois', 'Fioul', 'Gaz', 'Électrique', 'Électrique, Bois'], dtype='object', name='chauffage_energie')

La variable catégorielle 'type_annonceur' contient les modalités pr et pa pour PRO et PARTICULIER.

La variable catégorielle 'typedebien' contient les modalités m, a, l, mn et an pour Maison, Appartement, Lot, (Maison ?) et (Appartement ?).

La variable catégorielle 'typedetransaction' contient les modalités l, lt, v, vp et pi pour Location, (lt : type de location ?), Vente, (vp : type de vente?) et (pi ?).

La variable numérique 'etage' correspond à l'étage où se situe le bien. Pour les locations : les valeurs vont de -2 à 999 mais la médiane est de 1 et la moyenne de 1.8. Pour les ventes : les valeurs vont de -1 à 999 mais la médiane est de 0 et la moyenne de 1.3.

La variable numérique 'surface' correspond à la superficie en m² du bien. Pour les locations : les valeurs vont de 7 à 450 avec une médiane de 62 et une moyenne de 65.1. Pour les ventes : les valeurs vont de 2 à 980 avec une médiane de 95 et une moyenne de 108.3.

La variable numérique 'surface_terrain' correspond à la superficie en m² du terrain associé au bien. Pour les locations : les valeurs vont de 1.7 à 10198 avec une médiane de 455 et une moyenne de 564.1. Pour les ventes : les valeurs vont de 1 à 431600 avec une médiane de 589 et une moyenne de 1017.4.

La variable numérique 'nb_pieces' correspond aux nombres de pièces dans le bien. Pour les locations : les valeurs vont de 1 à 13 avec une médiane de 3 et une moyenne de 2.8. Pour les ventes : les valeurs vont de 1 à 43 avec une médiane de 4 et une moyenne de 4.5.

La variable numérique 'prix_bien' correspond au prix en € du bien (loyer ou vente). Pour les locations : les valeurs vont de 1 à 3287 avec une médiane de 660 et une moyenne de 714. Pour les ventes : les valeurs vont de 1^3 à 3.3^6 avec une médiane de 2.4^5 et une moyenne de 2.7^5.

La variable numérique 'prix_maison' correspond au prix en € de la maison associé au bien (loyer ou vente). Pour les locations : NaN. Pour les ventes : les valeurs vont de 86000 à 514999 avec une médiane de 199620 et une moyenne de 213134.4. (Il y a très peu de valeurs dans le jeu de données)

La variable numérique 'prix_terrain' correspond au prix en € du terrain associé au bien (loyer ou vente). Pour les locations : NaN. Pour les ventes : les valeurs vont de 1 à 320000 avec une médiane de 87000 et une moyenne de 89757.5. (Il y a très peu de valeurs dans le jeu de données)

La variable numérique 'mensualiteFinance' correspond au remboursement mensuel du bien (vente). Pour les locations : 0. Pour les ventes : les valeurs vont de 0 à 7715 avec une médiane de 0 et une moyenne de 13.2. (Il y a beaucoup de 0 dans le jeu de données).

La variable numérique 'balcon' correspond au nombre de balcons dans le bien. Pour les locations : les valeurs vont de 0 à 12 avec une médiane de 0 et une moyenne de 0.18. Pour les ventes : les valeurs vont de 0 à 8 avec une médiane de 0 et une moyenne de 0.28. (Il y a beaucoup de 0 dans le jeu de données).

La variable numérique 'eau' correspond à ? (nb salle d'eau ?) dans le bien. Pour les locations : les valeurs vont de 0 à 3 avec une médiane de 0 et une moyenne de 0.28. Pour les ventes : les valeurs vont de 0 à 10 avec une médiane de 0 et une moyenne de 0.28.

La variable numérique 'bain' correspond à ? (nb salle de bain ?) dans le bien. Pour les locations : les valeurs vont de 0 à 127 avec une médiane de 1 et une moyenne de 0.55. Pour les ventes : les valeurs vont de 0 à 12 avec une médiane de 1 et une moyenne de 0.68.

La variable catégorielle 'dpeL' correspond à la classe de Diagnostic de Performance Energetique du bien et contient les modalités 'D', 'E', 'C', 'VI', '0', 'B', 'F', 'NS', 'A', 'G'.

La variable numérique 'dpeC' correspond au score de Diagnostic de Performance Energetique du bien. Pour les locations : les valeurs vont de 0 à 939 avec une médiane de 188 et une moyenne de 195. Pour les ventes : les valeurs vont de 0 à 988 avec une médiane de 196 et une moyenne de 205.

La variable numérique 'mapCoordonneesLatitude' correspond à la latitude de la localisation GPS du bien. Pour les locations : les valeurs vont de 43.6 à 49.3. Pour les ventes : les valeurs vont de 47.4 à 48.3. => Département du Haut-Rhin

La variable numérique 'mapCoordonneesLongitude' correspond à la longitude de la localisation GPS du bien. Pour les locations : les valeurs vont de 0.07 à 7.58. Pour les ventes : les valeurs vont de 6.85 à 7.61. => Département du Haut-Rhin

La variable catégorielle 'annonce_exclusive' correspond à l'exclusivité commerciale d'une agence sur la location ou vente du bien et contient les modalités 0, 'Oui', 'Non'.

La variable numérique 'nb_etages' correspond au nombres d'étages présents dans le bien. Pour les locations : les valeurs vont de 0 à 169 mais la médiane est de 3 et la moyenne de 3.15. Pour les ventes : les valeurs vont de 0 à 36 mais la médiane est de 2 et la moyenne de 2.87.

La variable 'parking' est vide.

La variable numérique 'places_parking' correspond au nombres de place de parking associées au bien. Pour les locations : les valeurs vont de 0 à 216 mais la médiane est de 1 et la moyenne de 5.58. Pour les ventes : les valeurs vont de 0 à 35 mais la médiane est de 2 et la moyenne de 1.98.

La variable catégorielle 'cave' correspond à la présence ou non d'une cave associée au bien et contient les modalités True, False.

La variable catégorielle 'exposition' correspond à l'exposition du bien et contient les modalités '0', 'Sud', 'Sud-Ouest', 'Est', 'Ouest', 'Nord-Ouest', 'Sud-Est', 'Est-Ouest', 'sud-ouest', 'Nord-Sud', 'Sud-Est-Ouest', 'Nord', 'Nord-Est', 'sud-est', 'traversant', 'Nord Sud Est Ouest', 'Nord Est Ouest', 'nord,est', 'Nord Sud Ouest', 'sud,est', 'sud,ouest', 'Nord Sud Est', 'nord-est / sud-ouest', 'Traversant-Est-Ouest', 'nord-ouest / sud-est', 'nord,ouest', 'traversant nord-sud', 'sud est', 'sud ouest', 'SUD', '360°', 'Nord-ouest', 'SUD OUEST', 'EST/OUEST', 'nord ouest'. (Beaucoup de 0 dans les jeux de données)

La variable catégorielle 'ges_class' correspond à la classe d'émission de Gaz à Effet de Serre du bien et contient les modalités 'D', 'E', 'C', 'B', 'A', 'F', 'VI', 'G', 'NS'.

La variable numérique 'annee_construction' correspond à l'année de construction du bien. Pour les locations : les valeurs vont de 1001 à 2023 mais la médiane est de 1985 et la moyenne de 1973. Pour les ventes : les valeurs vont de 971 à 2025 mais la médiane est de 1978 et la moyenne de 1969.

La variable numérique 'nb_toilettes' correspond au nombres de toillettes présentes dans le bien. Pour les locations : les valeurs vont de 0 à 14 mais la médiane est de 1 et la moyenne de 1.13. Pour les ventes : les valeurs vont de 0 à 15 mais la médiane est de 1 et la moyenne de 1.49.

La variable numérique 'nb_terraces' correspond au nombres de terraces présentes dans le bien. Pour les locations : les valeurs vont de 0 à 3 mais la médiane est de 1 et la moyenne de 0.53. Pour les ventes : les valeurs vont de 0 à 14 mais la médiane est de 1 et la moyenne de 0.99.

La variable catégorielle 'videophone' correspond à la présence ou non d'un videophone dans le bien et contient les modalités True, False.

La variable catégorielle 'porte_digicode' correspond à la présence ou non d'un digicode sur la porte d'entrée du bien et contient les modalités True, False.

La variable numérique 'surface_balcon' correspond à la superficie en m² du/des balcon(s) du bien. Pour les locations : les valeurs vont de 1 à 31 avec une médiane de 8 et une moyenne de 8.39. Pour les ventes : les valeurs vont de 1 à 801 avec une médiane de 10 et une moyenne de 12.48. (Beaucoup de valeurs vides)

La variable catégorielle 'ascenseur' correspond à la présence ou non d'un ascenseur dans le bien et contient les modalités True, False. (Beaucoup de valeurs vides)

La variable numérique 'nb_logements_copro' correspond au nombres de logements dans la même copropriété que le bien. Pour les locations : seulement 9 valeurs. Pour les ventes : les valeurs vont de 0 à 3442 mais la médiane est de 24 et la moyenne de 53. (Beaucoup de valeurs vides)

La variable numérique 'charges_copro' correspond au montant en € des charges de copropriété du bien. Pour les locations : 0 valeurs. Pour les ventes : les valeurs vont de 0 à 2.5^6 mais la médiane est de 1.2^3 et la moyenne de 1.8^3. (Beaucoup de valeurs vides)

La variable catégorielle 'chauffage_energie' correspond au type de chauffage dans le bien et contient les modalités Bois', 'Bois, Fioul', 'Bois, Électrique', 'Fioul', 'Fioul, Bois', 'Gaz', 'Gaz, Bois', 'Gaz, Électrique', 'Électrique', 'Électrique, Bois'. (Beaucoup de valeurs vides)

TODO :
- 'chauffage_systeme',
- 'chauffage_mode',
- 'categorie_annonceur',
- 'logement_neuf',
- 'duree_int',
- 'typedebien_lite',
- 'date',
- 'INSEE_COM',
- 'IRIS',
- 'CODE_IRIS',
- 'TYP_IRIS_x',
- 'TYP_IRIS_y',
- 'GRD_QUART',
- 'UU2010',
- 'REG',
- 'DEP'
- 'loyer_m2_median_n6',
- 'nb_log_n6',
- 'taux_rendement_n6',
- 'loyer_m2_median_n7',
- 'nb_log_n7',
- 'taux_rendement_n7',
- 'prix_m2_vente'

### Variables selectionnées pour la première analyse

Après une prise de connaissance des contenus des fichiers csv, de la liste des colonnes, de leurs valeurs, distribution et répartition - nous avons fait le choix de nous focaliser sur les variables suivantes :
- localisation
- surface
- etages
- type de bien
- nb de pièces

## DataViz'

#### Données de locations

#### Données de ventes